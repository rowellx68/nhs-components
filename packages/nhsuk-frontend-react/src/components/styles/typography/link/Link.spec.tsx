import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  Default as DefaultStory,
  NoVisitedState as NoVisitedStateStory,
} from './Link.stories';

const Default = composeStory(DefaultStory, meta);
const NoVisitedState = composeStory(NoVisitedStateStory, meta);

it('should render the Link component', () => {
  const { container } = render(<Default />);

  expect(container.querySelector('.nhsuk-link')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the Link component without visited state', () => {
  const { container } = render(<NoVisitedState />);

  expect(
    container.querySelector('.nhsuk-link--no-visited-state'),
  ).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
