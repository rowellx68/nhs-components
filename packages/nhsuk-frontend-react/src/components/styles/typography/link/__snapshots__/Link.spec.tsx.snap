// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the Link component 1`] = `
<div>
  <a
    class="nhsuk-link"
    href="#"
  >
    Link
  </a>
</div>
`;

exports[`should render the Link component without visited state 1`] = `
<div>
  <a
    class="nhsuk-link nhsuk-link--no-visited-state"
    href="#"
  >
    Link nhsuk-link--no-visited-state
  </a>
</div>
`;
